lib/rubocop/cop/rspec/empty_example_group.rb

```diff
@@ -60,15 +60,28 @@ module RSpec
       class EmptyExampleGroup < Cop
         MSG = 'Empty example group detected.'
 
-        def_node_search :contains_example?, <<-PATTERN
+        def_node_matcher :contains_example?, <<-PATTERN
           {
-            #{(Examples::ALL + Includes::ALL).send_pattern}
-            (send _ #custom_include? ...)
+            #{Examples::ALL.block_pattern}
+            #{ExampleGroups::ALL.block_pattern}
+            #{Includes::ALL.send_pattern}
+            (send nil? #custom_include? ...)
+            (block (send _ :each) _ #contains_example?)
+            (begin <(block (send _ :each) _ #contains_example?) ...>)
           }
         PATTERN
 
         def on_block(node)
-          return unless example_group?(node) && !contains_example?(node)
+          return unless example_group?(node)
+
+          body =
+            if node.body&.begin_type?
+              node.body.each_child_node
+            else
+              [node.body].each
+            end
```

Comment:
I guess this can be simplified and implemented in node pattern, but it's way too late around here for me to crack it.
Neither `return if contains_example?(node)` nor `return if node.children.any?(&method(:contains_example?))` worked with the existing node pattern - some specs fail.

Reply:
I guess the else could be just `[node.body]`
easier to understand, provides `any?`